sdks/typescript/src/apache_beam/runners/direct_runner.ts

```diff
@@ -194,24 +197,29 @@ class DirectGbkOperator implements operators.IOperator {
     );
     const windowingStrategy =
       context.descriptor.windowingStrategies[inputPc.windowingStrategyId];
-    // TODO: (Cleanup) Check or implement triggers, etc.
     if (
       windowingStrategy.mergeStatus !== runnerApi.MergeStatus_Enum.NON_MERGING
     ) {
-      throw new Error("Non-merging WindowFn: " + windowingStrategy);
+      throw new Error("Unsupported non-merging WindowFn: " + windowingStrategy);
+    }
+    if (
+      windowingStrategy.outputTime !== runnerApi.OutputTime_Enum.END_OF_WINDOW
+    ) {
+      throw new Error(
+        "Unsupported windowing output time: " + windowingStrategy
+      );
     }
```

Comment:
Should we use `CombinePerKeyPrecombineOperator.checkSupportsWindowing` here?

Reply:
No, as these may evolve separately. (On that note, I started merging these two implementations, but it muddied what the pure GBK one was doing so I left that here for pedagogical purposes.)